### PR TITLE
Numerical linearization test

### PR DESCRIPTION
**Pull request by crupp2**
Test for OpenModelica/OMCompiler#532.

### CODE REVIEW

**Comment by lochel** [comment]
The empty merge commit is probably not needed.

**Comment by lochel** [comment]
Should we add some more tests for the numeric linearization? We could just run some of the symbolic ones additionally with the numeric approach.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-03-28_12-07-19](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/2089/).

**Comment by lochel** [comment]
@crupp2 The changes don’t break anything. There is just a small issue with your test case. I think we can merge this once the test is updated and some of the symbolic linearization tests were adapted.

**Comment by crupp2** [comment]
@lochel that is a good idea. Should I add numeric to each existing file or add a new file for each test with _numeric.mos? What about data recovery, how much of that should I test?

Unstable test suite is ok? Or is that because of my failed test?

File: openmodelica/linearization/simNonlinear_numeric.mos [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+loadFile("linear_p.nonlinsys.mo");
+list(linear_p_nonlinsys);
+simulate(linear_p_nonlinsys);
+getErrorString();
+readFile("linear_p_nonlinsys.log"); // Check that output log is emtpy
+
+// Result:
+// true
+// record SimulationResult
+//     resultFile = "S:/Scratch/p.nonlinsys_res.mat",
```
Comment (lochel) [inline openmodelica/linearization/simNonlinear_numeric.mos]: The test suit is unstable due to this and a similar line further down. Just replace it with `resultFile = "p.nonlinsys_res.mat",`.

**Comment by lochel** [comment]
Maybe we could have both approaches in one file to make it easier to verify that they give the same result. @adrpo what do you think?

**Comment by crupp2** [comment]
Added numeric tests for all of the test cases. Also validated that they match the symbolic and that the data recovery matrices are correct.

**Comment by lochel** [comment]
Great. I just triggered again a Hudson job to see if everything is fine now.

Can you remove the empty merge commits in order to keep things clean and simple? If you don’t know how to do that, I can just clean up your pull request and merge it.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-03-29_20-56-52](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/2125/).

**Comment by crupp2** [comment]
Ug, I just tried this out locally and they worked. Is this running against the master omc? Maybe something changed?

**Comment by crupp2** [comment]
Those empty merge commits were just me living in a svn world. Unfortunately, I cannot figure out how to get rid of them, so I'll need your help.

**Comment by vruge** [comment]
git rebase --preserve-merges -i HEAD~3

**Comment by lochel** [comment]
There are several ways of doing this with git, e.g. rebase your changes and then push them using the `--force` flag.
```
$ git rebase -i HEAD~2
$ git push crupp2 numerical_linearization_test --force
```
Please be careful using `--force`, since it will change the history.

**Comment by crupp2** [comment]
I think that worked. Thanks for the help.

**Comment by crupp2** [comment]
FYI, the failed test was due to a difference in how Windows vs. Linux is processing the system() call. I was testing on Windows and I assume that Hudson is Linux. Tests should be good for another go.

**Comment by lochel** [comment]
I've squashed your last two commits and triggered Hudson to push your changes.

**Pull request closed without merging**